nuitka/optimizations/BytecodeDemotion.py

```diff
@@ -66,14 +66,17 @@ def getModuleImportableFilesHash(full_name):
     return result_hash.hexdigest()
 
 
-def _getCacheFilename(full_name, module_importables_hash):
+def _getCacheFilename(full_name, module_importables_hash, extension):
     module_cache_dir = os.path.join(getCacheDir(), "module-imports-cache")
     makePath(module_cache_dir)
 
     return os.path.join(
-        module_cache_dir, "%s-%s.txt" % (full_name.asString(), module_importables_hash)
+        module_cache_dir, "%s-%s.%s" % (full_name.asString(), module_importables_hash, extension)
     )
 
+def _loadBytecodeFromCache(cache_filename):
```

Comment:
Using `getFileContents` from `nuitka.utils.FileOperations` will respect locking. This might be important later down the road, if Nuitka runs in more processes, and avoids the code.